key/src/test/java/software/amazon/kms/key/UpdateHandlerTest.java

```diff
@@ -191,10 +195,7 @@ public void handleRequest_SoftFailAccessDenied() {
             .updateKeyPolicy(eq(proxy), eq(proxyKmsClient), eq(KEY_MODEL_PREVIOUS), eq(KEY_MODEL),
                 eq(callbackContext)))
             .thenReturn(inProgressEvent);
-        when(eventualConsistencyHandlerHelper.waitForChangesToPropagate(eq(inProgressEvent)))
-            .thenReturn(inProgressEvent);
-
-        // Throw a CfnAccessDeniedException from updateKeyTags so we can test that it is soft failed
+        // Remove kms:TagResource permission. This throws CfnAccessDeniedException
```

Comment:
nit: I word reword this comment. We're not removing kms:TagResource permission to throw CfnAccessDeniedException. We're instead throwing CfnAccessDeniedException to mimic removing kms:TagResource permission

Reply:
+1 recommend "Throw CfnAccessDeniedException to mock no kms:TagResource permission" like you used below